### PR TITLE
feat(simulation): adds operations for streamer module

### PR DESCRIPTION
**Pull request by danwt**
## Description

----

Also fixes an invariant on sponsorship

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

----

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: x/streamer/keeper/keeper.go [hunk 1]
```diff
@@ -151,6 +168,9 @@ func (k Keeper) TerminateStream(ctx sdk.Context, streamID uint64) error {
 	if err != nil {
 		return err
 	}
+	if stream.IsFinishedStream(ctx.BlockTime()) {
```
Comment (mtsitrin) [inline x/streamer/keeper/keeper.go:171]: isn't it covered by the code below?
```
else {
	return fmt.Errorf("stream %d is not active or upcoming", streamID)
}
```

File: x/streamer/proposal_handler.go [hunk 2]
```diff
@@ -27,52 +27,45 @@ func NewStreamerProposalHandler(k keeper.Keeper) govtypes.Handler {
...
 
 // HandleCreateStreamProposal is a handler for executing a passed community spend proposal
-func HandleCreateStreamProposal(ctx sdk.Context, k keeper.Keeper, p *types.CreateStreamProposal) error {
-	_, err := k.CreateStream(ctx, p.Coins, p.DistributeToRecords, p.StartTime, p.DistrEpochIdentifier, p.NumEpochsPaidOver, p.Sponsored)
-	if err != nil {
-		return err
-	}
-	return nil
+func HandleCreateStreamProposal(ctx sdk.Context, k keeper.Keeper, p *types.CreateStreamProposal) (err error) {
+	defer func() {
```
Comment (mtsitrin) [inline x/streamer/proposal_handler.go:31]: why u need to run it in defer?

File: x/streamer/keeper/keeper.go [hunk 3]
```diff
@@ -28,6 +29,7 @@ type Keeper struct {
 	ik         types.IncentivesKeeper
 	sk         types.SponsorshipKeeper
 
+	schema collections.Schema // set later
```
Comment (mtsitrin) [inline x/streamer/keeper/keeper.go:32]: can u add doc for it's usage? very unclear

**Review by mtsitrin — APPROVED** [review]

**Comment by danwt** [comment]
Not convinced this is useful tbh
GOnna conv to draft

**Pull request closed without merging**